crates/noirc_abi/src/input_parser/mod.rs

```diff
@@ -183,20 +186,52 @@ fn parse_str_to_field(value: &str) -> Result<FieldElement, InputParserError> {
     if value.starts_with("0x") {
         FieldElement::from_hex(value).ok_or_else(|| InputParserError::ParseHexStr(value.to_owned()))
     } else {
-        value
-            .parse::<i128>()
+        BigUint::from_str_radix(value, 10)
             .map_err(|err_msg| InputParserError::ParseStr(err_msg.to_string()))
-            .map(FieldElement::from)
+            // Note that we're reducing here. Should we reject inputs that don't fit in the field?
```

Comment:
Yeah I think its better to be more restrictive initially unless there is a need -- this means that the input needs to be canonical

Reply:
Ah, I left this as I thought the hex wasn't performing this check. Looks like it does so will reduce here as well.